src/lager_handler_watcher.erl

```diff
@@ -194,5 +192,17 @@ reinstall_on_runtime_failure_test_() ->
         ]
     }.
 
+pop_until(String, Fun) ->
```

Comment:
Well, I pass `lists:flatten` for one invocation, but wrap it with further logic (`string:substr`) for another, or do I misunderstand your question?

That's all I expect for now, although the pattern seems generally useful.